# Spark Streaming Testing

## Import Libraries and Initiate SparkContext

In [1]:
# We release the SparkContext if it exists.
try:
    sc
except:
    pass ;
else:
    sc.stop()

# Now handle initial import statements
from pyspark import SparkConf, SparkContext
from pyspark.streaming import StreamingContext

# Create new Spark Configuration (port numbers might need to be adjusted from defaults.)
myconf = SparkConf()
myconf.setMaster('local[*]')
myconf.setAppName("INFO490 SP17 W14-NB1: Your Name")
myconf.set('spark.executor.memory', '1g')

# Create and initialize a new Spark Context
sc = SparkContext(conf=myconf)
ssc = StreamingContext(sc, 1)

# Display Spark version information, which also verifies SparkContext is active
print("\nSpark version: {0}".format(sc.version))



Spark version: 2.1.1


## Import Dataset to RDD

In [4]:
distFile = sc.textFile('./food_business_ids.pkl')

In [5]:
distFile.count()

1228

1228 Observations, a nice small set to start with.

## Stream Data 

In [6]:
# Create a DStream that will connect to hostname:port, like localhost:9090
lines = ssc.socketTextStream("localhost", 9090)

In [18]:
# Split each line into words
words = lines.flatMap(lambda line: line.split(" "))

In [19]:
# Count each word in each batch
pairs = words.map(lambda word: (word, 1))
wordCounts = pairs.reduceByKey(lambda x, y: x + y)

# Print the first ten elements of each RDD generated in this DStream to the console
wordCounts.pprint()

In [20]:
ssc.start()             # Start the computation
ssc.awaitTermination()  # Wait for the computation to terminate

-------------------------------------------
Time: 2017-06-03 19:11:21
-------------------------------------------

-------------------------------------------
Time: 2017-06-03 19:11:22
-------------------------------------------

-------------------------------------------
Time: 2017-06-03 19:11:23
-------------------------------------------

-------------------------------------------
Time: 2017-06-03 19:11:24
-------------------------------------------

-------------------------------------------
Time: 2017-06-03 19:11:25
-------------------------------------------

-------------------------------------------
Time: 2017-06-03 19:11:26
-------------------------------------------

-------------------------------------------
Time: 2017-06-03 19:11:27
-------------------------------------------

-------------------------------------------
Time: 2017-06-03 19:11:28
-------------------------------------------

-------------------------------------------
Time: 2017-06-03 19:11:29
----------

KeyboardInterrupt: 